server/middleware/users.coffee

```diff
@@ -52,3 +52,19 @@ module.exports =
       { multi: true }
     )
     next()
+
+  remainTeacher: wrap (req, res, next) ->
+    res.status(200).send()
+
+  removeOwnedClassrooms: wrap (req, res, next) ->
+    userID = mongoose.Types.ObjectId(req.user.id)
+    yield Classroom.remove({ ownerID: userID }, false)
+    next()
+
+  setStudentRole: wrap (req, res, next) ->
+    userID = mongoose.Types.ObjectId(req.user.id)
+    yield User.update({ _id: userID }, { $set: { "role": "student" } })
+    next()
+
+  becomeStudent: wrap (req, res, next) ->
```

Comment:
I think it would make more sense to put all the logic of this endpoint in one function. If there are actions, like `removeOwnedClassrooms`, which are shared across endpoints, we could add it as a method to `User`, so you would call `yield req.user.removeOwnedClassrooms()`.